deployer/src/deployment/queue.rs

```diff
@@ -52,7 +55,7 @@ pub async fn task(
         let storage_manager = storage_manager.clone();
         let queue_client = queue_client.clone();
 
-        tokio::spawn(async move {
+        tasks.spawn(async move {
```

Comment:
Very good point, I think this can happen yes. If the queue never ends (e.g. in the case of a long running process), the task set will never be awaited, and the memory will keep growing with each deployment. 

I wonder how this could be fixed. I see two solutions, I don't know if there are some simpler ones:
1. Spawn a thread that will be responsible for running `join_next` + a signal once we're done with the queue to notify that thread that the next time `join_next` returns `None` (i.e. that the set is empty), it must return.
2. Instead of doing a while loop on `recv.recv()`, use a `select!` between `recv.recv()`, `tasks.join_next()`, and the else case where both returned `None`.

I think the second one is the cleanest, let me know what you think !

Reply:
Ideally, we would use something like 2. However, this doesn't seem to crash anything in our stress testing so we think about merging it without the extra handling for now. There is also a hard limit imposed on the container memory which will most probably crash the user's deployer, but not affect the rest of the system. Also, we will move away from this one deployer-per-user's project architecture soon, so it is not critical to address this.